source/llassetgen/source/Packing.cpp

```diff
@@ -0,0 +1,88 @@
+#include <llassetgen/Packing.h>
+
+#include <algorithm>
+#include <tuple>
+
+namespace llassetgen {
+    Packing::Packing(Vec2<PackingSizeType> _atlasSize) : atlasSize{_atlasSize} {}
+
+    namespace impl_packing {
+        bool ShelfNextFitPacker::packNext(Vec2<PackingSizeType> rectSize) {
+            if (currentShelfSize.x + rectSize.x > packing.atlasSize.x) {
+                openNewShelf();
+                if (rectSize.x > packing.atlasSize.x) {
+                    return false;
+                }
+            }
+
+            if (usedHeight + rectSize.y > packing.atlasSize.y) {
+                return false;
+            }
+
+            store(rectSize);
+            return true;
+        }
+
+        bool ShelfNextFitPacker::packNextRotatable(Vec2<PackingSizeType> rectSize) {
+            PackingSizeType minSide, maxSide;
```

Comment:
We discourage multiple declarations per line.
Further, we discourage declaration without definition / initialization.

Reply:
I refactored the code to avoid the `rotated` variable being left uninitialized. I personally prefer `std::tie` over manual tuple unpacking (which would avoid uninitialized variables), at least without C++17/structured bindings.